modules/core/04-channel/keeper/keeper.go

```diff
@@ -653,8 +671,41 @@ func (k Keeper) HasPruningSequenceStart(ctx sdk.Context, portID, channelID strin
 	return store.Has(host.PruningSequenceStartKey(portID, channelID))
 }
 
-// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than or equal to the pruning sequence.
+// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than pruning sequence end.
 // The number of packet acknowledgements pruned is equal to limit. Pruning only occurs after a channel has been upgraded.
-func (Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit, pruningSequence uint64) error {
+//
+// The pruningSequence keeps track of the packet acknowledgement that can be pruned next. When the pruning sequence reaches
+// the last send sequence, pruning is complete.
+func (k Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit uint64) error {
+	if !k.HasPruningSequenceStart(ctx, portID, channelID) {
+		// indicates that a channel upgrade has not been performed
+		return errorsmod.Wrapf(types.ErrPruningSequenceStartNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+	pruningSequenceStart := k.GetPruningSequenceStart(ctx, portID, channelID)
```

Comment:
Should we also do here `pruningSequenceStart, found := k.GetPruningSequenceStart(...`?